Cmdline/Action/KSP.cs

```diff
@@ -248,6 +292,113 @@ private int AddInstall(AddOptions options)
             }
         }
 
+        private int CloneInstall(CloneOptions options)
+        {
+            log.Info("Cloning the KSP instance: " + options.nameOrPath);
+
+            // Parse all options
+            string instanceNameOrPath = options.nameOrPath;
+            string new_name = options.new_name;
+            string new_path = options.new_path;
+
+            // Try instanceNameOrPath as name and search the registry for it.
+            if (Manager.HasInstance(instanceNameOrPath))
+            {
+                CKAN.KSP[] listOfInstances = Manager.Instances.Values.ToArray();
+                foreach (CKAN.KSP instance in listOfInstances)
+                {
+                    if (instance.Name == instanceNameOrPath)
+                    {
+                        // Found it, now clone it.
+                        try
+                        {
+                            Manager.CloneInstance(instance, new_name, new_path);
+                        }
+                        catch (NotKSPDirKraken kraken)
+                        {
+                            // Two possible reasons:
+                            // First: The instance to clone is not a valid KSP instance.
+                            // Only occurs if user manipulated directory and deleted files/folders
+                            // which CKAN searches for in validity test.
+
+                            // Second: Something went wrong adding the new instance to the registry,
+                            // most likely because the newly created directory is not valid.
+
+                            User.RaiseError(kraken.ToString());
+                            log.Error(kraken);
+                            return Exit.ERROR;
+                        }
+                        catch (IOException e)
+                        {
+                            // The new path is not empty
+                            // The exception contains a message to inform the user.
+
+                            User.RaiseError(e.ToString());
+                            log.Error(e);
+                            return Exit.ERROR;
+                        }
+                        break;
+                    }
+                }
+            }
+            // Try to use instanceNameOrPath as a path and create a new KSP object.
+            // If it's valid, go on.
+            else if (new CKAN.KSP(instanceNameOrPath, new_name, User) is CKAN.KSP instance && instance.Valid)
+            {
+                // Same as above.
+                try
+                {
+                    Manager.CloneInstance(instance, new_name, new_path);
+                }
+                catch (NotKSPDirKraken kraken)
```

Comment:
Is there any way we could consolidate these `catch` blocks with the identical ones above? Any future fixes to these areas would have to be made in multiple places.

Reply:
Yep, changed in next commit.